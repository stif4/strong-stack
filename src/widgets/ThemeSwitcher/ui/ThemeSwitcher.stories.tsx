import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { ThemeSwitcher } from './ThemeSwitcher';

export default {
    title: 'shared/ThemeSwitcher',
    component: ThemeSwitcher,
} satisfies Meta<typeof ThemeSwitcher>;

const Template: StoryFn<typeof ThemeSwitcher> = (args) => <ThemeSwitcher {...args} />;

export const Normal = Template.bind({});
Normal.args = {};
