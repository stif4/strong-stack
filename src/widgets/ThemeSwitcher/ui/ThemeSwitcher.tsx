import { classNames } from 'shared/helpers/classNames/classNames';
import LightIcon from 'shared/assets/icons/theme-light.svg';
import DarkIcon from 'shared/assets/icons/theme-dark.svg';
import { Button, ButtonTheme } from 'shared/ui/Button/Button';
import { useTheme } from 'app/providers/themeProviders';
import { Theme } from 'app/providers/themeProviders/ui/ThemeContext';

interface ThemeSwitcherProps {
    className?: string;
}

export const ThemeSwitcher = ({ className }: ThemeSwitcherProps) => {
    const { theme, toggleTheme } = useTheme();

    return (
        <Button
            theme={ButtonTheme.CLEAR}
            className={classNames('', {}, [className])}
            onClick={toggleTheme}
        >
            {theme === Theme.DARK ? <DarkIcon /> : <LightIcon />}
        </Button>
    );
};
