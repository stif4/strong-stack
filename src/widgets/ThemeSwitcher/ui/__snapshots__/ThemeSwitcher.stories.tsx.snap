// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared/ThemeSwitcher Normal smoke-test 1`] = `
<div class="app dark">
  <button class="src-shared-ui-Button-Button-module__button src-shared-ui-Button-Button-module__clear">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="40"
         height="41"
         fill="none"
    >
      <path fill="#0115C6"
            d="M10.848 1.044A12.87 12.87 0 0 0 4.931 11.88c0 4.571 2.387 8.57 5.976 10.837C4.852 22.717 0 17.87 0 11.88a10.83 10.83 0 0 1 3.177-7.662 10.85 10.85 0 0 1 7.671-3.174"
      >
      </path>
      <path fill="#0115C6"
            d="m14.97 15.9-2.288-1.439-2.209 1.537.671-2.6-2.15-1.636 2.683-.177.887-2.542 1.006 2.502 2.683.06-2.072 1.714zM21.48 8.787l-2.92-1.832-2.84 1.97.828-3.35-2.741-2.088 3.451-.236L18.402 0l1.322 3.211 3.412.06-2.663 2.226zM33.669 4l2.82 2.817L5.78 37.493l-2.82-2.817zM33.53 23.702a10.83 10.83 0 0 1-3.177 7.662 10.854 10.854 0 0 1-14.1 1.066L31.42 17.279a10.8 10.8 0 0 1 2.11 6.423M33.393 31.346l2.268-5.457L40 30.893zM35.66 21.574l-2.248-5.477 6.588.472zM24.852 36.666l5.464-2.266-.474 6.6zM15.05 34.4l5.463 2.266-4.99 4.314z"
      >
      </path>
    </svg>
  </button>
</div>
`;
